pkg/proc/bininfo.go

```diff
@@ -2273,7 +2273,7 @@ func (bi *BinaryInfo) findType(name string) (godwarf.Type, error) {
 	return godwarf.ReadType(image.dwarf, ref.imageIndex, ref.offset, image.typeCache)
 }
 
-func (bi *BinaryInfo) findTypeExpr(expr ast.Expr) (godwarf.Type, error) {
+func (bi *BinaryInfo) FindTypeExpr(expr ast.Expr) (godwarf.Type, error) {
```

Comment:
Functions like `findType`, `expandPackagesInType`, and `findArrayType` are all private functions, and it can be quite challenging to implement these functions externally.

Reply:
The only call to FindTypeExpr in the repository you linked that I can find is this:

```
				t, err := parser.ParseExpr("runtime.specialfinalizer")
				if err != nil {
					return err
				}
				spty, _ := special.bi.FindTypeExpr(t)
```

You don't need `FindTypeExpr` for this, you can just use `godwarf.ReadType`.